Challenge2rmd file1.Rmd

```diff
@@ -0,0 +1,105 @@
+---
+title: "Exercise 12_Challenge2"
+author: "Bhavana"
+date: "November 18, 2017"
+output: html_document
+---
+
+## R Markdown and regular expressions
+This is an R Markdown document that describes regular expression to match patterns. Regular expressions are extremely useful in extracting information from text such as code, log files, spreadsheets, or even documents.Below is the code in R to match some patterns.
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r}
+#load library for string r and i
+library(stringr)
+library(stringi)
+
+```
+
+
+
+a)times after noon, but before midnight when reported in 24-hour or military format (e.g. 15:30)
+#create a file with some military time format data
+#load data
+#write regex for time = "[1][2][:][0-9][1-9]|[1][3-9][:][0-9]{2}|[2][0-9][:][0-9]{2}"
+
+
+```{r militarytimeformat}
+
+data = scan(file="militarytimeformat.txt",what=character(),sep="\n")
+data
+```
+
+
+```{r}
+grep("([1][2][:][0-9][1-9]|[1][3-9][:][0-9]{2}|[2][0-9][:][0-9]{2})", data)
```

Comment:
could simplify and test more efficiently by making the expression its own argument. For example, 

regex1="(1[2-9]|2[0-3]):[0-5][0-9]"

testset=c('09:30','01:55','cat','15:45','22:15')

str_detect(testset,regex1)